src/huggingface_hub/keras_mixin.py

```diff
@@ -199,7 +207,18 @@ def save_pretrained_keras(
         with open(path, "w") as f:
             json.dump(config, f)
 
-    _create_model_card(model, save_directory, plot_model, task_name)
+    metadata = {}
+    if isinstance(tags, list):
+        metadata["tags"] = tags
+    elif isinstance(tags, str):
+        metadata["tags"] = [tags]
+    if "task_name" in locals():
```

Comment:
In lines 194 to 199 you can maybe do 

```py
task_name = model_save_kwargs.pop("task_name", None) # This means the default will be None
if task_name is not None:
   warnings.warn(
            "`task_name` input argument is removed. Pass `tags` instead.",
            FutureWarning,
        )
```

and similarly, here you won't need to check `locals()`, which seems a bit hacky, and you can just do
`if task_name is not None`.

Reply:
it doesn't work, it says "`task_name` local variable mentioned before assignment" that's why I had to do it. I'll check if I mention it anywhere else.

Reply:
Yes, you get the error because `task_name` is created within an if. My suggestion is to have this outside all the ifs. 

```
os.makedirs(save_directory, exist_ok=True)

task_name = model_save_kwargs.pop("task_name", None) # This means the default will be None
if task_name is not None:
   warnings.warn(
            "`task_name` input argument is removed. Pass `tags` instead.",
            FutureWarning,
        )
...
```

Reply:
ah thanks!